[FCA-04] Implement main screen

matxowy opened the pull request:
- Implemented main screen
- Added colors and themes from theme builder to support dark mode
- Added basic dimensions in dimens.xml
- Disabled jetifier to improve building time
- Downgraded material version to 1.7 due to problem with rendering with TextInputLayout

matxowy committed:
[FCA-04]

- Implemented main screen
- Added colors and themes from theme builder to support dark mode
- Added basic dimensions in dimens.xml
- Disabled jetifier to improve building time
- Downgraded material version to 1.7 due to problem with rendering with TextInputLayout

msniezek-Sii commented on app/src/main/res/layout/main_screen_fragment.xml:
```diff
@@ -0,0 +1,78 @@
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/small_margin"
+        android:textAppearance="@style/TextAppearance.Material3.BodyLarge"
```
Powinieneś użyć raczej czegoś takiego
`android:textAppearance="?attr/textAppearanceBodyLarge"`

matxowy replied:
O nie wiedziałem, że textAppearance jest też dostępny przez ?attr. Dobrze wiedzieć i będę używał 😄

msniezek-Sii commented on app/src/main/res/layout/main_screen_fragment.xml:
```diff
@@ -0,0 +1,78 @@
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="@dimen/small_margin"
+        android:textAppearance="@style/TextAppearance.Material3.BodyLarge"
```
Jak wyżej
`android:textAppearance="?attr/textAppearanceBodyLarge"`

matxowy replied:
Poprawione ☺️

msniezek-Sii commented on app/src/main/res/layout/main_screen_fragment.xml:
```diff
@@ -0,0 +1,78 @@
+        app:layout_constraintTop_toBottomOf="@id/tv_choose_category"
+        app:layout_constraintWidth_max="@dimen/max_constraint_width_for_textview">
+
+        <AutoCompleteTextView
```
com.google.android.material.textfield.MaterialAutoCompleteTextView

matxowy replied:
Ah to przyzwyczajenie 😅 Poprawione

msniezek-Sii commented on app/src/main/res/layout/main_screen_fragment.xml:
```diff
@@ -0,0 +1,78 @@
+            android:inputType="none" />
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textview.MaterialTextView
```
Może zamiast tworzenia oddzielnego TextView można by użyć `android:hint` w TextInputLayout

matxowy replied:
Dałem ten TextView bo wydawało mi się, że mało będzie na tym screenie strasznie, a tu zrobi sztuczny tłum no, ale ładniej raczej wygląda z hintem także zmieniłem 😄

msniezek-Sii commented on app/src/main/res/layout/main_screen_fragment.xml:
```diff
@@ -0,0 +1,78 @@
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/til_spinner_categories"
+        style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox.ExposedDropdownMenu"
```
`@style/Widget.Material3.TextInputLayout.OutlinedBox.ExposedDropdownMenu`

matxowy replied:
Poprawione ☺️

msniezek-Sii commented on app/src/main/res/layout/main_screen_fragment.xml:
```diff
@@ -0,0 +1,78 @@
+        style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox.ExposedDropdownMenu"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="?attr/colorOnSecondary"
```
Tak tego nie robimy ;) Żeby zmieniać kolor jakiegokolwiek widgetu w taki sposób (a tak w ogóle to `app:boxBackgroundColor="?attr/colorOnSecondary"`) musi być bardzo dobre uzasadnienie a tu takiego nie widzę. Jeśli chcesz, żeby ExposedDropdownMenu miały jasne tło (a faktycznie lepiej wygląda z jasnym :) ) to trzeba to zrobić na poziomie stylu tak jak to robimy z kolorami 
`<style name="Theme.FlashcardsApp" parent="Theme.Material3.Light.NoActionBar">
        <item name="colorPrimary">@color/md_theme_light_primary</item>`
albo tworzymy nowy styl na bazie `@style/Widget.Material3.TextInputLayout.OutlinedBox.ExposedDropdownMenu"` i jego używamy w xml. 
Generalnie lepiej to nadpisać w stylu jeśli się da. Do poczytania: https://github.com/material-components/material-components-android/blob/master/docs/components/Menu.md#exposed-dropdown-menus

matxowy replied:
Dzięki za wytłumaczenie i podrzucenie artykułu, będę pamiętał na przyszłość 😄 Poprawione ☺️

matxowy committed:
[FCA-04]

- Added style to themes for textInputStyle to change the color of the background
- Changed writing convention from snake case to camel case
- Text from TextView moved to hint
- Removed vertical guideline
